### ISSUE TITLE
Python API: numpy array as parameter for get_derivative

### ISSUE THREAD

**Issue by neXyon** [darioizzo/audi#16]
The Python API of get_derivative should accept numpy arrays as parameters, not only lists.

For example:

```
prod = np.sum([gdual(0, _, 1) for _ in 'xyz'])

index = np.zeros(prod.symbol_set_size)
index[2] = 1

print(prod.get_derivative(index))
```

should do the same as

```
prod = np.sum([gdual(0, _, 1) for _ in 'xyz'])

index = list(np.zeros(prod.symbol_set_size))
index[2] = 1

print(prod.get_derivative(index))
```

**Comment by darioizzo** [comment]
I agree. pybind11 automatically converts c++ std::containers in lists, not numpy array. I think @bluescarni was looking into elegant ways to have numpy arrays in the API too (across AuDi, piranha, pagmo etc. )

**Comment by bluescarni** [comment]
@darioizzo @neXyon I need to contact the pybind11 maintainer to see if the solution I have is the way to go or not. My hunch is that I'd like pybind11 to use numpy by default when working on std vectors, but this might be unfeasible or undesirable in the general case (e.g., there's not much advantage in having a numpy array of ``object``s if you are converting a std::vector of something else than floating point values).
